### DISCUSSION TITLE
Create a workflow to handle qr codes in exhibitions

### DISCUSSION THREAD

**Discussion by rondlg** [fieldmuseum/EMu-Documentation#1048] [FAQs] [answered]
related to exhibitions QR code ZD tickets:

(Elshtain poems)
- https://fieldmuseum.zendesk.com/agent/tickets/51893 
- https://fieldmuseum.zendesk.com/agent/tickets/53326 
- https://fieldmuseum.zendesk.com/agent/tickets/53330

(NA Hall 8)
- https://fieldmuseum.zendesk.com/agent/tickets/52656

Request is to set up a method for handling web pages that launch from physical qr codes that are in the galleries'. In these 2 examples the need is for a user to be able to scan a qr code, pull up the matching text and see any translations of the same text on a web page.

**Comment by rondlg** [comment]
**Draft workflow**

Each qr code is a mm record in EMu (the short.io url is attached as associated mm with Relationship = Published URL). The qr code mm record is reverse attached to the main narrative for the poem which is itself attached to one or more associated narratives holding the translations.
 
Attached is a possible layout for a detail page that has narratives attached to it.
![mm_fieldmuseum_org_with_narrative](https://user-images.githubusercontent.com/702709/156417358-6d041a81-9589-4cb6-bd2e-948b8e8e1fd6.jpg)
 
Example mm irn = 2315982

**Comment by rondlg** [comment]
From @magpiedin 

narratives irns 7590 & 7591 & mm 2315982 are set up on Ross/skeletally  (& Event 8796 for the home page is on Ross but not in Solr)

**Comment by rondlg** [comment]
@magpiedin I have this straightened out now. 

There are 2 translation asks but a single workflow will cover both:

EveTypeOfEvent = "Web Project - QR Code"

**web project translation event records:**
- event irn 8797 -> Poetry in Residence translations (multiple exhibitions/ multiple locations)
-- This is Marie's question/ticket
-- This is the best example for @peteherbst to work with. All the mm records are hooked up, the translations are in the narratives module and there is a pdf attached to each narrative.

- event irn 8796 -> Hall 8 translations (single exhibition/multiple locations)
-- This is Taylor's question/ticket
-- The records are hooked up, but we don't have the actual text from exhibitions for the narratives yet so we can't finish this up.

**Comment by magpiedin** [comment]
Event irn 8797 & attached Narratives + MM are now in + attached & set to publish in Ross

Updates for event irn 8796 are on their way by end of day, too in case exh text comes through before end of week.

**Comment by magpiedin** [comment]
fyi -- for QR-code Multimedia records -- Resource Subtype references [this acsubtype controlled vocab](https://ac.tdwg.org/subtype/#3-term-index) in case an existing value works in place of 'QR Code' -- 'Graphic' or 'Icon'?

**Comment by rondlg** [comment]
Whatever makes sense. Where should we explicitly state that the mm record is a qr-code?

**Comment by rondlg** [comment]
**Draft workflow**

There should be an event record for onsite translations that go with qr-codes (EveTypeOfEvent = "Web Project - QR code"). This may or may not be attached to a parent Exhibition event (for example Poet in Residence does not necessarily need an exhibition parent as the content is distributed across the museum). The narratives with the content should be attached to this event NOT the parent. 

pj.fieldmuseum.org layout events where EveTypeOfEvent = "Web Project - QR code" with reverse attached narratives.

![pj_fieldmuseum_org_with_narrative](https://user-images.githubusercontent.com/702709/156460080-475f43cf-1516-4de0-b6cb-faea1944447f.jpg)

**Comment by magpiedin** [comment]
have to wait til next week / web refresh for the rest of the narratives tied to event 8796; tried to get them in for half a day and i'm either missing some attachments or messed something up with NarNarrative / character encoding / Ross or both

recommend Multimedia / DetSubjects_tab = 'QR Code' for now if that's ok; I'll ask audubon core group about / how to go about adding 'QR code' (&/or [barcode](https://schema.org/Barcode)) to the vocab

**Comment by rondlg** [comment]
Workflow document is here: https://docs.google.com/document/d/1CQbu3GGMaDcK-7UoJPXGJF-zNKqxBlI2wbsTrf3CMVc

**Comment by rondlg** [comment]
development request: https://github.com/fieldmuseum/EMu-Events/issues/4

Example: https://pj.fieldmuseum.tech/exhibition/1e42b1d2-936c-4e9e-883b-aacd3caa7957

**Comment by rondlg** [comment]
Record Structure diagram: https://docs.google.com/presentation/d/1V4UxsU6m84XUACY9G8qxbuRoNh0SOiYIIR3FPiGuMQE

**Comment by rondlg** [comment]
The second use case is when there is/are audio file(s) to go with the text.

**Comment by rondlg** [comment]
Updates to the qr-code page requested for travelling Apsaalooke Women and Warriors 

- ZD case is https://fieldmuseum.zendesk.com/agent/tickets/54555 

1) re-order author (NarAuthorsRef_tab.NamFullName) below text

- example enarratives irn 7592
- - emultimedia irn 2316026

2) if NarNarrative is NULL don't display the expander plus

- example enarratives irn 7592 (without)
- - emultimedia irn 2316026
- example enarratives irn 7611 (with)
- - emultimedia irn 2312235

3) If NarContributorRef_tab is not NULL AND NarContributorRole_tab = Translator 
   then display 'Translator: ' + NarContributorRef_tab .NamFullName 

- example enarratives irn 7610
- - emultimedia irn 2312236

Also, to discuss the possibilities for:

4) allow the 1st pdf attached to the narrative to be available for download/opening?

- example enarratives irn 7611
- - emultimedia irn 2312235

5) display of the 1st attached jpg image to the narrative

- example emultimedia irn 2352600

Something like this:
![image](https://user-images.githubusercontent.com/702709/167041836-d8fbaa40-9bd2-4a09-85b9-e72f9ad973ea.png)

**Comment by peteherbst** [comment]
@rondlg For the above is there an example MM URL I can work from? I didn't see anything in the ticket.

**Comment by rondlg** [comment]
emultimedia irns added to previous comment: https://github.com/fieldmuseum/EMu-Documentation/issues/174#issuecomment-1119130182

**Comment by peteherbst** [comment]
Please test these out. It's on the dev site so you need to be on VPN to test.

1. [Author below text](https://mm.fieldmuseum.tech/83e61e17-f265-4fb5-90d6-90a03d90696b)
2. [Expander without](https://mm.fieldmuseum.tech/83e61e17-f265-4fb5-90d6-90a03d90696b), [expander with](https://mm.fieldmuseum.tech/27350ae4-e8d4-4992-903c-005a01a6a87c)
3. [Translator](https://mm.fieldmuseum.tech/f9328712-24a5-4213-883d-52b6cad790cf)
4. [First PDF ](https://mm.fieldmuseum.tech/27350ae4-e8d4-4992-903c-005a01a6a87c)
5. [First image](https://mm.fieldmuseum.tech/d375a0dd-4869-47a4-9adb-6aa826af378b)

**Comment by rondlg** [comment]
Genius - those all work.

**Comment by peteherbst** [comment]
I've pushed the changes to prod (i.e. the .org website). Let me know if you need anything else on this!

**Comment by rondlg** [comment, marked as answer]
# [Here is the final workflow](https://docs.google.com/document/d/1CQbu3GGMaDcK-7UoJPXGJF-zNKqxBlI2wbsTrf3CMVc)

**Before you start you will need to gather the following:**
- The original text and pdfs of the original text/document
- The text and pdfs of any translations
- Permission to use the text/translations.
- QR codes as svg files
- The urls to go with the QR codes
- The event numbers of any exhibitions associated with the translations.
- If you are to have audio with the text you will need .wav or mp3 files
